### PR TITLE
[DOC:] New content and new page: Discover main page and multiple data sources

### LINKED ISSUE #1710: [DOC] New main page: Discover OpenSearch Dashboards
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [X] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Create Discover main page

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by vagimeli**
Signed-off-by: vagimeli <vagimeli@amazon.com>

### Description
Add new main page

### Issues Resolved
#1710 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ariamarble — APPROVED** [review]
LGTM

File: _dashboards/discover/multi-data-sources.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+1. In the Dashboards console, select **Index Patterns > Create index pattern**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-5.png" width=500 alt="Index pattern user interface">
+
+2. Select **Use external data source connection**.
+3. Start typing in the Search data sources field to search for the data source you created earlier and then select the data source and **Next step**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-6.png" width=500 alt="Index pattern search user interface">
+
+4. Add an **Index pattern name** to define the index pattern and then select **Next step**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-7.png" width=500 alt="Index pattern define user interface">
```
Comment (kristenTian) [inline _dashboards/discover/multi-data-sources.md:68]: This screen shot is not accurate, due to cannot use input `sample` to create index pattern. You can try use `opensearch*` instead

File: _dashboards/discover/multi-data-sources.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+---
+layout: default
+title: Multiple data sources
+parent: Discover
+nav_order: 5
+---
+
+Multiple data sources is an experimental feature released in OpenSearch 2.4. We do not recommend using it in a production environment. For updates on the feature’s progress or to leave feedback on improving the feature, see the [multiple data sources git issue](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2448).
```
Comment (zhongnansu) [inline _dashboards/discover/multi-data-sources.md:8]: wrong link? why it's linked to a merged PR

File: _dashboards/discover/multi-data-sources.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-12.jpg" width=500 alt="Example dashboard using data visualizations from many data sources">
+
+You have explored the data sources experimental feature. We look forward to your feedback on how we can improve this feature ahead of its release for production use.
+
+## Limitations
+
+The following limitations apply to the 2.4 release of this experimental feature:
+
+* The multiple data sources currently only works with index pattern based visualizations
+* Visualization types TSVB, Vega and Vega-Lite, and timeline are not supported.
```
Comment (zhongnansu) [inline _dashboards/discover/multi-data-sources.md:126]: external plugins(such as [gant chart](https://github.com/opensearch-project/dashboards-visualizations)), and non-visualization plugins (such as dev console) are also not supported. Should we mention here or in the codebase README.

Comment (zhongnansu) [reply _dashboards/discover/multi-data-sources.md:126]: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2455
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1990

File: _dashboards/discover/multi-data-sources.md (hunk 2 above)
Comment (kristenTian) [reply _dashboards/discover/multi-data-sources.md:8]: Did you meant the forum link? https://forum.opensearch.org/t/feedback-experimental-feature-connect-to-external-data-sources/11144

File: _dashboards/discover/multi-data-sources.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+## Try out multiple data sources in your local environment
+
+This tutorial uses a preconfigured data source and index pattern, and you aren’t required to configure settings. However, you’ll need to enable the multiple data sources setting in the configuration file before before getting started with exploring this feature.
+{: .note }
+
+Multiple data sources is an experimental feature that can't be deployed into production, but you can try it out with a sample data source and a sample index pattern. Before getting started, you must first edit the YAML configuration. The following section provides the steps to enable the multiple data sources feature.
+
+## Edit the YAML configuration to enable multiple data sources 
+
+Dashboards is configured in the cluster settings, and multiple data sources is disabled by default. To enable multiple data sources, you need to edit the configuration in `opensearch_dashboards.yml` and then restart the cluster. To enable multiple data sources:
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:26]: Do we want to style these features somehow – "multiple data sources"? "Discover" is bold in the first mention of it above. "Multiple data sources" is such a common string of words that alone it doesn't feel like a feature and gets lost in context, as in, "multiple data sources is disabled...". I know it's crossing our rule. But can we give them initial caps?

File: _dashboards/discover/multi-data-sources.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+4. Add an **Index pattern name** to define the index pattern and then select **Next step**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-7.png" width=500 alt="Index pattern define user interface">
+
+5. Select an option for the **Time field**, then choose **Create index pattern**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-8.jpg" width=500 alt="Index pattern time field user interface">
+
+## Search data
+
+Before getting started with searching your data, set up the time filter. The sample index pattern used for this tutorial contains time-based data. You can set a time filter that displays only the data within a specified time range and you can select the time filter to change the time range or select a specific time range in the histogram.
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:75]: Suggestion: delete "...getting started with..." in first sentence, or, "Before you start searching for data...".

File: _dashboards/discover/multi-data-sources.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+### Use the time filter
+
+1. In the Dashboards console, select **Discover** and confirm index pattern being used is `opensearch_dashboards_sample_data_ecommerce`.
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:79]: Is there a reason for dropping the articles on "index pattern" in line 79 and "time field" in line 80? In line 81 it's "the time field".

File: _dashboards/discover/multi-data-sources.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+2. Select calendar icon to change time field. The default is **Last 15 minutes**.
+3. Change the time field to **Last 7 days** and select **Refresh**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-9.png" alt="Time filter user interface">
+
+4. To set the start and end times, select the bar next to the time filter. In the popup, select **Absolute**, **Relative**, or **Now**, then specify the required options.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-10.png" width=300 alt="Start and end times user interface">
+
+### Select a time range from the histogram
+
+To select a time range for the histogram, you can choose one of the following:
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:89]: "... you can **do** one of the following:"?

File: _dashboards/discover/multi-data-sources.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-12.jpg" width=500 alt="Example dashboard using data visualizations from many data sources">
+
+You have explored the data sources experimental feature. We look forward to your feedback on how we can improve this feature ahead of its release for production use.
+
+## Limitations
+
+The following limitations apply to the 2.4 release of this experimental feature:
+
+* The multiple data sources currently only works with index pattern based visualizations
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:125]: Again, "multiple data sources ... works with..." rubs me the wrong way.

**Review by cwillum — APPROVED** [review]
Nice work. See comments. Otherwise, good to go.

File: _dashboards/discover/multi-data-sources.md (hunk 8 above)
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:125]: Revised.

File: _dashboards/discover/multi-data-sources.md (hunk 4 above)
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:26]: I'm tracking with your comment. I revised throughout to "the multiple data sources feature" to clarify where appropriate (as in this para.)

File: _dashboards/discover/multi-data-sources.md (hunk 1 above)
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:70]: replaced image

File: _dashboards/discover/multi-data-sources.md (hunk 3 above)
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md]: Added

File: _dashboards/discover/multi-data-sources.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+You can add multiple data sources to a single dashboard. OpenSearch Dashboards allows you to dynamically manage data sources, create index patterns based on those data sources, and execute queries against a specific data source and then combine visualizations in one dashboard.
+
+In this tutorial we provide the steps for enabling the data sources setting in Dashboards; adding credentials, data source connections, and index patterns; and combining visualizations in a single dashboard.
+
+## Try out Data Sources in your local environment
+
+This tutorial uses a preconfigured data source and index pattern, and you aren’t required to configure settings. However, you’ll need to enable the multiple data sources setting in the configuration file before before getting started with exploring this feature.
+{: .note }
+
+Multiple data sources is an experimental feature that can't be deployed into production, but you can try it out with a sample data source and a sample index pattern. Before getting started, you must first edit the YAML configuration. The following section provides the steps to enable the feature.
```
Comment (ariamarble) [inline _dashboards/discover/multi-data-sources.md:22]: Should we maybe use language to have people try it and to leave feedback? Maybe like "we encourage you to try out / test this feature and provide feedback" etc?

File: _dashboards/discover/multi-data-sources.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+Before you start searching for data, set up the time filter. The sample index pattern used for this tutorial contains time-based data. You can set a time filter that displays only the data within a specified time range and you can select the time filter to change the time range or select a specific time range in the histogram.
+
+### Use the time filter
+
+1. In the Dashboards console, select **Discover** and confirm the index pattern being used is `opensearch_dashboards_sample_data_ecommerce`.
+2. Select the calendar icon to change the time field. The default is **Last 15 minutes**.
+3. Change the time field to **Last 7 days** and select **Refresh**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-9.png" alt="Time filter user interface">
+
+4. To set the start and end times, select the bar next to the time filter. In the popup, select **Absolute**, **Relative**, or **Now**, then specify the required options.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:86]:
```suggestion
4. To set the start and end times, select the bar next to the time filter. In the popup, select **Absolute**, **Relative**, or **Now** and then specify the required options.
```

File: _dashboards/discover/multi-data-sources.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-5.png" width=500 alt="Index pattern user interface">
+
+2. Select **Use external data source connection**.
+3. Start typing in the Search data sources field to search for the data source you created earlier and then select the data source and **Next step**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-6.png" width=500 alt="Index pattern search user interface">
+
+4. Add an **Index pattern name** to define the index pattern and then select **Next step**.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-7.png" width=500 alt="Index pattern define user interface">
+
+5. Select an option for the **Time field**, then choose **Create index pattern**.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:72]:
```suggestion
5. Select an option for the **Time field** and then choose **Create index pattern**.
```

File: _dashboards/discover/multi-data-sources.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+4. To set the start and end times, select the bar next to the time filter. In the popup, select **Absolute**, **Relative**, or **Now**, then specify the required options.
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-10.png" width=300 alt="Start and end times user interface">
+
+### Select a time range from the histogram
+
+To select a time range for the histogram, you can do one of the following:
+
+* Select the bar that represents the time range you want to zoom in on.
+* Select the bar and drag to view a specific time range. You must start the selection with the cursor over the background of the chart—the cursor changes to a plus sign when you hover over a valid start point.
+* Select the dropdown, then select an interval.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:95]:
```suggestion
* Select the dropdown and then select an interval.
```

File: _dashboards/discover/multi-data-sources.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-11.jpg" alt="Histogram user interface">
+
+## Create visualizations
+
+Follow these steps to learn how to connect your visualizations in a single dashboard.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:101]:
```suggestion
Follow these steps to learn how to connect your visualizations in a single dashboard:
```

File: _dashboards/discover/multi-data-sources.md [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+1. In the Dashboards console, select **Visualize** > **Create visualization**.
+2. Select the visualization type. For this tutorial, select **Line**.
+3. Choose a source. For this tutorial, select the index pattern `opensearch_dashboards_sample_data_ecommerce`.
+4. Under **Buckets**, select **Add > X-axis**.
+5. In the **Aggregation** field, select **Date Histogram** and then **Update**.
+6. Select **Save** and add file name. This tutorial uses preconfigured visualizations, so you won’t be able to save your visualization.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:108]: It looks like "a" or "the" should precede "file name".

File: _dashboards/discover/multi-data-sources.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+
+## Connect visualizations in a single dashboard
+
+1. In the Dashboards console, select **Dashboard > Create dashboard**. 
+2. Select **Add an existing** and then select the data you want to add.
+3. Select **Save** and add dashboard name in the **Title field**. This tutorial uses preconfigured dashboards, so you won’t be able to save your dashboard.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:114]: "add the dashboard name"?

File: _dashboards/discover/multi-data-sources.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+4. Click into dashboard white space left of **Add panels** to view the visualizations in a single dashboard.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:115]:
```suggestion
4. Click into the dashboard white space to the left of **Add panels** to view the visualizations in a single dashboard.
```

Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:115]: Is there better wording than "click into"?

File: _dashboards/discover/multi-data-sources.md [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+Follow these steps to learn how to connect your visualizations in a single dashboard.
+
+1. In the Dashboards console, select **Visualize** > **Create visualization**.
+2. Select the visualization type. For this tutorial, select **Line**.
+3. Choose a source. For this tutorial, select the index pattern `opensearch_dashboards_sample_data_ecommerce`.
+4. Under **Buckets**, select **Add > X-axis**.
+5. In the **Aggregation** field, select **Date Histogram** and then **Update**.
+6. Select **Save** and add file name. This tutorial uses preconfigured visualizations, so you won’t be able to save your visualization.
+
+## Connect visualizations in a single dashboard
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:110]: Can we include a brief sentence between the heading and the list that introduces the list?

File: _dashboards/discover/multi-data-sources.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+1. In the Dashboards console, select **Dashboard > Create dashboard**. 
+2. Select **Add an existing** and then select the data you want to add.
+3. Select **Save** and add dashboard name in the **Title field**. This tutorial uses preconfigured dashboards, so you won’t be able to save your dashboard.
+4. Click into dashboard white space left of **Add panels** to view the visualizations in a single dashboard.
+
+Your dashboard might look like this:
+
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-12.jpg" width=500 alt="Example dashboard using data visualizations from many data sources">
+
+You have explored the data sources experimental feature. We look forward to your feedback on how we can improve this feature ahead of its release for production use.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:121]:
```suggestion
You have now explored the data sources experimental feature. We look forward to your feedback on how we can improve this feature ahead of its release for production use.
```

File: _dashboards/discover/multi-data-sources.md [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+
+## Limitations
+
+The following limitations apply to the 2.4 release of this experimental feature:
+
+* The multiple data sources feature is supported for index pattern-based visualizations only.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:127]:
```suggestion
* The multiple data sources feature is supported for index-pattern-based visualizations only.
```

File: _dashboards/discover/multi-data-sources.md [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+* Visualization types Time Series Visual Builder (TSVB), Vega and Vega-Lite, and timeline are not supported.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:128]:
```suggestion
* The visualization types Time Series Visual Builder (TSVB), Vega and Vega-Lite, and timeline are not supported.
```

File: _dashboards/discover/multi-data-sources.md [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+
+Your dashboard might look like this:
+
+<img src="{{site.url}}{{site.baseurl}}/images/multi-data-sources-12.jpg" width=500 alt="Example dashboard using data visualizations from many data sources">
+
+You have explored the data sources experimental feature. We look forward to your feedback on how we can improve this feature ahead of its release for production use.
+
+## Limitations
+
+The following limitations apply to the 2.4 release of this experimental feature:
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:125]: "OpenSearch 2.4"?

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes, and let me know if you have any questions. Thanks!

File: _dashboards/discover/multi-data-sources.md [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+
+To enable the feature:
+
+1. Navigate to your Dashboards home directory; for example, in Docker `/usr/share/opensearch-dashboards`.
+2. Open your local copy of the Dashboards configuration file, `opensearch_dashboards.yml`. If you don't have a copy, get one from GitHub: [`opensearch_dashboards.yml`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml).
+3. Set `data_source.enabled: false` to  `data_source.enabled: true` and save the configuration.
+4. Restart the Dashboards container.
+5. Verify the feature configuration settings were created and configured properly by opening Dashboards on [http://localhost:5601](http://localhost:5601/) and viewing the **Stack Management** console. **Data Sources `Experimental`** will appear in the sidebar. Alternatively, you can open on [http://localhost:5601/app/management/opensearch-dashboards/dataSources](http://localhost:5601/app/management/opensearch-dashboards/dataSource).
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:34]: Revised.

File: _dashboards/discover/multi-data-sources.md [hunk 23]
```diff
@@ -0,0 +1,134 @@
...
+---
+
+Multiple data sources is an experimental feature released in OpenSearch 2.4. We do not recommend using it in a production environment. For updates on the feature’s progress or to leave feedback on improving the feature, see the [OpenSearch Forum discussion](https://forum.opensearch.org/t/feedback-experimental-feature-connect-to-external-data-sources/11144).
+{: .note }
+
+# Multiple data sources
+
+You can add multiple data sources to a single dashboard. OpenSearch Dashboards allows you to dynamically manage data sources, create index patterns based on those data sources, and execute queries against a specific data source and then combine visualizations in one dashboard.
+
+In this tutorial we provide the steps for enabling the data sources setting in Dashboards; adding credentials, data source connections, and index patterns; and combining visualizations in a single dashboard.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:15]: Agreed. Changes made globally.

File: _dashboards/discover/multi-data-sources.md [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+
+## Try out Data Sources in your local environment
+
+This tutorial uses a preconfigured data source and index pattern, and you aren’t required to configure settings. However, you’ll need to enable the multiple data sources setting in the configuration file before before getting started with exploring this feature.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:19]: Revised per previous comment.

File: _dashboards/discover/multi-data-sources.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:22]: Revised.

**Comment by vagimeli** [comment]
Ready for release with OpenSearch 2.4

**Comment by noCharger** [comment]
@vagimeli Thanks for creating the doc! Just curious, is this by design to put the doc under `discover` section? Since OpenSearch Dashboard also has a feature with same name, I am not sure if people will get confused. 

Screen Shot 2022-11-16 at 8 29 23 PM

Screen Shot 2022-11-16 at 8 29 41 PM

**Comment by vagimeli** [comment]
@noCharger We've been restructuring the doc site. Below is the working structure. Will you share your thoughts where multiple data sources best fits in the structure?  Thanks! 

- Get started
- Quickstart
- Core concepts
- Discover
- Dashboard
- Visualize
- Maps
- Reporting

**Pull request merged**